File: windowsforms/matching-game/net45/cs/MatchingGame.Logic/GameSettings.cs
using System;
using Microsoft.Win32;

namespace MatchingGame.Logic
{
    public sealed class GameSettings
    {
        private const string KeyPath = "Software\\AdventureWorks\\MatchingGame";
        public static readonly GameSettings Instance = Load();

        private int _bestScore;

        private GameSettings()
        {
        }

        public int BestScore
        {
            get => _bestScore;
            private set
            {
                _bestScore = value;
                Save();
            }
        }

        public void UpdateScore(int newScore)
        {
            if (BestScore == 0 || newScore < BestScore)
                BestScore = newScore;
        }

        private static GameSettings Load()
        {
            var result = new GameSettings();

            using (var key = Registry.CurrentUser.OpenSubKey(KeyPath, false))
            {
                if (key != null)
                {
                    var value = key.GetValue(nameof(BestScore));
                    if (value != null)
                        result.BestScore = Convert.ToInt32(value);
                }
            }

            return result;
        }

        private void Save()
        {
            using (var key = Registry.CurrentUser.CreateSubKey(KeyPath))
            {
                key.SetValue(nameof(BestScore), BestScore);
            }
        }
    }
}

File: windowsforms/matching-game/net8-windows/vb/MatchingGame.Logic/GameSettings.vb
Imports Microsoft.Win32

Public NotInheritable Class GameSettings

    Private Const KeyPath As String = "Software\\AdventureWorks\\MatchingGame"
    Public Shared ReadOnly Instance As GameSettings = Load()

    Private _bestScore As Integer

    Private Sub New()

    End Sub

    Public Property BestScore As Integer
        Get
            Return _bestScore
        End Get
        Set
            _bestScore = Value
            Save()
        End Set
    End Property

    Public Sub UpdateScore(newScore As Integer)
        If BestScore = 0 Or newScore < BestScore Then
            BestScore = newScore
        End If
    End Sub

    Private Shared Function Load() As GameSettings

        Dim result = New GameSettings()

        Using key = My.Computer.Registry.CurrentUser.OpenSubKey(KeyPath, False)
            If key IsNot Nothing Then
                Dim value = key.GetValue(NameOf(BestScore))
                If value IsNot Nothing Then
                    result.BestScore = CInt(value)
                End If
            End If
        End Using

        Return result

    End Function

    Private Sub Save()
        Using key = My.Computer.Registry.CurrentUser.OpenSubKey(KeyPath, True)
            key.SetValue(NameOf(BestScore), BestScore)
        End Using
    End Sub

End Class
